--- acpica-unix-20120420/source/compiler/aslerror.c.bero	2012-05-14 14:04:20.568179913 +0200
+++ acpica-unix-20120420/source/compiler/aslerror.c	2012-05-14 14:04:33.078180123 +0200
@@ -375,7 +375,8 @@ AePrintException (
 
                             else while (RActual && SourceByte && (SourceByte != '\n'))
                             {
-                                fwrite (&SourceByte, 1, 1, OutputFile);
+                                if(fwrite (&SourceByte, 1, 1, OutputFile) < 1)
+                                    fprintf(stderr, "Write failure: %s\n", strerror(errno));
                                 RActual = fread (&SourceByte, 1, 1, SourceFile);
                             }
                         }
--- acpica-unix-20120420/source/compiler/dttemplate.c.bero	2012-05-14 14:06:51.641183237 +0200
+++ acpica-unix-20120420/source/compiler/dttemplate.c	2012-05-14 14:08:36.247185598 +0200
@@ -411,11 +411,13 @@ DtCreateOneTemplate (
         AcpiOsPrintf (" */\n\n");
         if (ACPI_COMPARE_NAME (Signature, ACPI_SIG_DSDT))
         {
-            fwrite (TemplateDsdt, sizeof (TemplateDsdt) -1, 1, File);
+            if(fwrite (TemplateDsdt, sizeof (TemplateDsdt) -1, 1, File) < 1)
+                fprintf(stderr, "DSDT Write failure: %s\n", strerror(errno));
         }
         else if (ACPI_COMPARE_NAME (Signature, ACPI_SIG_SSDT))
         {
-            fwrite (TemplateSsdt, sizeof (TemplateSsdt) -1, 1, File);
+            if(fwrite (TemplateSsdt, sizeof (TemplateSsdt) -1, 1, File) < 1)
+                fprintf(stderr, "SSDT Write failure: %s\n", strerror(errno));
         }
         else if (ACPI_COMPARE_NAME (Signature, ACPI_SIG_FACS))
         {
--- acpica-unix-20120420/source/os_specific/service_layers/osunixxf.c.bero	2012-05-14 14:13:00.137191465 +0200
+++ acpica-unix-20120420/source/os_specific/service_layers/osunixxf.c	2012-05-14 14:13:57.496192740 +0200
@@ -436,7 +436,9 @@ AcpiOsGetLine (
             return (AE_BUFFER_OVERFLOW);
         }
 
-        scanf ("%1c", &Temp);
+        if(scanf ("%1c", &Temp) < 0)
+            fprintf(stderr, "AcpiOsGetLine failed: %s\n", strerror(errno));
+
         if (!Temp || Temp == '\n')
         {
             break;
--- acpica-unix-20120420/source/tools/acpibin/abcompare.c.bero	2012-05-14 14:15:00.618194143 +0200
+++ acpica-unix-20120420/source/tools/acpibin/abcompare.c	2012-05-14 14:15:24.472194676 +0200
@@ -782,7 +782,8 @@ AbExtractAmlFile (
 
                     /* Write the converted (binary) byte */
 
-                    fwrite (&Value, 1, 1, FileOutHandle);
+                    if(fwrite (&Value, 1, 1, FileOutHandle)<1)
+                        fprintf(stderr, "Couldn't write binary data: %s\n", strerror(errno));
                     Count++;
                 }
             }
--- acpica-unix-20120420/source/tools/acpisrc/asfile.c.bero	2012-05-14 14:27:04.435210235 +0200
+++ acpica-unix-20120420/source/tools/acpisrc/asfile.c	2012-05-14 14:28:01.201211497 +0200
@@ -877,7 +877,8 @@ AsPutFile (
     /* Write the buffer to the file */
 
     FileSize = strlen (FileBuffer);
-    write (DestHandle, FileBuffer, FileSize);
+    if(write (DestHandle, FileBuffer, FileSize)<0)
+        fprintf(stderr, "Couldn't write to %s: %s\n", Pathname, strerror(errno));
 
     close (DestHandle);
 
--- acpica-unix-20120420/source/tools/acpisrc/asmain.c.bero	2012-05-14 14:29:23.803213334 +0200
+++ acpica-unix-20120420/source/tools/acpisrc/asmain.c	2012-05-14 14:30:45.817215211 +0200
@@ -249,7 +249,8 @@ AsExaminePaths (
     if (!AsStricmp (Source, Target))
     {
         printf ("Target path is the same as the source path, overwrite?\n");
-        scanf ("%c", &Response);
+        if(scanf ("%c", &Response)<1)
+            fprintf(stderr, "Can't read response: %s\n", strerror(errno));
 
         /* Check response */
 
@@ -266,7 +267,8 @@ AsExaminePaths (
         if (!Status)
         {
             printf ("Target path already exists, overwrite?\n");
-            scanf ("%c", &Response);
+            if(scanf ("%c", &Response)<1)
+                fprintf(stderr, "Can't read response: %s\n", strerror(errno));
 
             /* Check response */
 
